src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php

```diff
@@ -21,61 +21,87 @@
  * This class was generated by the Doctrine ORM. Add your own custom
  * repository methods below.
  */
-class CollectionRepository extends EntityRepository implements CollectionRepositoryInterface
+class CollectionRepository extends NestedTreeRepository implements CollectionRepositoryInterface
 {
+    /**
+     * {@inheritdoc}
+     */
+    public function findCollectionById($id)
+    {
+        $result = $this->findCollectionSet($id, 0);
+
+        if(sizeof($result) === 0){
+            return null;
+        }
+
+        return $result[0];
+    }
 
     /**
      * {@inheritdoc}
      */
-    public function findCollectionById($id, $asArray = false)
+    public function findCollectionSet($id, $depth = 0)
     {
         try {
-            $qb = $this->createQueryBuilder('collection')
-                ->leftJoin('collection.meta', 'collectionMeta')
-                ->leftJoin('collection.type', 'type')
-                ->leftJoin('collection.parent', 'parent')
-                ->leftJoin('collection.children', 'children')
-                ->leftJoin('children.media', 'childrenMedia')
-                ->leftJoin('collection.media', 'media')
+            $sql = sprintf(
+                'SELECT n, collectionMeta, collectionType, collectionParent, parentMeta, collectionMedia, collectionChildren
+                 FROM %s AS n
+                        LEFT JOIN n.meta AS collectionMeta
+                        LEFT JOIN n.type AS collectionType
+                        LEFT JOIN n.parent AS collectionParent
+                        LEFT JOIN n.children AS collectionChildren
+                        LEFT JOIN n.media AS collectionMedia
+                        LEFT JOIN collectionParent.meta AS parentMeta,
+                      %s AS p
+                 WHERE n.lft BETWEEN p.lft AND p.rgt
+                   AND (n.depth <= p.depth + :maxDepth OR collectionChildren.depth <= :maxDepthPlusOne)
+                   AND p.id = :id',
+                $this->_entityName,
+                $this->_entityName
+            );
```

Comment:
I think you said this kind of stuff is not natively supported by the Gedmo extensions, right?